cms/server/admin/handlers/dataset.py

```
@@ -1310,3 +1318,344 @@ def post(self, dataset_id, generator_id):
+        testcase.codename = new_codename
+        # Re-add to the collection (keyed by new codename)
+        dataset.testcases[new_codename] = testcase
+
+        # Update submission format for OutputOnly tasks
+        if dataset.active and dataset.task_type == "OutputOnly":
+            try:
+                task.set_default_output_only_submission_format()
+            except Exception as e:
+                raise RuntimeError(
+                    f"Couldn't create default submission format for task {task.id}, "
+                    f"dataset {dataset.id}") from e
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)